ydb/library/yql/providers/yt/lib/res_pull/res_or_pull.cpp

```diff
@@ -166,12 +199,17 @@ bool TSkiffExecuteResOrPull::WriteNext(TMkqlIOCache& specsCache, const NYT::TNod
         return false;
     }
 
+    // For now this method is used only for the nodes that are maps from column name
+    // to value. Feel free to extend it for other types (lists, for example) but
+    // make sure that the order of columns is correct.
+    YQL_ENSURE(rec.GetType() == NYT::TNode::EType::Map, "Expected map node");
+
     TStringStream err;
-    auto value = NCommon::ParseYsonNode(specsCache.GetHolderFactory(), rec, Specs.Outputs[tableIndex].RowType, Specs.Outputs[tableIndex].NativeYtTypeFlags, &err);
+    auto value = NCommon::ParseYsonNode(specsCache.GetHolderFactory(), rec, Specs.Outputs[0].RowType, Specs.Outputs[0].NativeYtTypeFlags, &err);
```

Comment:
In which queries might the problem here be?

Reply:
For example
```
select a, b, c from primary.`//tmp/t2`
union all
select a, b, c from primary.`//tmp/t1`

```

It has two input tables and a single output.

Reply:
I will add it as a test case to YTsaurus repo.